### ISSUE TITLE
"Connection is invalid" on listening for AtNotification

### ISSUE THREAD

**Issue by ralfeus** [atsign-foundation/at_client_sdk#741]
**Describe the bug**
While listening for `AtNotification` stream eventually sooner or later I'm getting an error:
```
Unhandled exception:
Exception: Connection is invalid
#0      BaseConnection.write (package:at_lookup/src/connection/base_connection.dart:46:7)
#1      Monitor._scheduleHeartbeat.<anonymous closure> (package:at_client/src/manager/monitor.dart:224:35)
#2      new Future.delayed.<anonymous closure> (dart:async/future.dart:424:39)
#3      Timer._createTimer.<anonymous closure> (dart:async-patch/timer_patch.dart:18:15)
#4      _Timer._runTimers (dart:isolate-patch/timer_impl.dart:398:19)
#5      _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:192:12)
```

**To Reproduce**
1. Run the code: 
```dart
  void listen() async {
    final atStream = AtClientManager.getInstance().notificationService.subscribe();
    await for (final notification in atStream) {
        ...
    }
  }
```
2. Wait 30-120 minutes

**Expected behavior**
Should listen for events indefinitely. 
- In case of network failure it might notify of one and try to reconnect but never stop listening.
- In case of long term network failure it should emit

**Comment by gkc** [comment]
Thanks @ralfeus.

**Comment by gkc** [comment]
Impl complete including new tests. Plan to release tomorrow Oct 18

**Comment by gkc** [comment]
Oops forgot about this. Will try to create PR this week along with some other PRs in at_client_sdk

**Comment by gkc** [comment]
Too much other higher priority work came up during PR49. Hopefully can complete in PR50

**Comment by gkc** [comment]
Added a couple more tests during PR50 so burned down ~1SP this sprint.

**Comment by gkc** [comment]
Will be closed by #822
